### PR TITLE
e2e groupby additional tests

### PR DESCRIPTION
**Pull request by itsmekumari**
This PR contains additional scenarios for e2e coverage of groupBy plugin

### CODE REVIEW

File: core-plugins/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -542,4 +542,12 @@ public static void createBucketWithRowDenormalizerTestFile() throws IOException,
       PluginPropertyUtils.pluginProp("rowDenoGcsCsvFile"));
     BeforeActions.scenario.write("RowDenormalizer bucket name - " + gcsSourceBucketName1);
   }
+
+  @Before(order = 1, value = "@GROUPBY_TEST")
+  public static void createBucketWithGroupByTest2File() throws IOException, URISyntaxException {
+    gcsSourceBucketName1 = createGCSBucketWithFile(PluginPropertyUtils.pluginProp("groupByGcsCsvTestFile"));
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:548]: Are we using constant names for source files?

Comment (itsmekumari) [reply core-plugins/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:548]: Yes, we are using constant names for source files and providing the path in plugin parameters file.

File: core-plugins/src/e2e-test/resources/errorMessage.properties [hunk 2]
```diff
@@ -29,3 +29,6 @@ errorLogsMessageDistinctInvalidFields=Spark program 'phase-1' failed with error:
 errorLogsMessageDistinctInvalidNumberOfPartitions=Spark program 'phase-1' failed with error: Unable to create config \
   for batchaggregator Distinct 'numPartitions' is invalid: Value of field class io.cdap.plugin.\
   batch.aggregator.AggregatorConfig.numPartitions is expected to be a number.. Please check the system logs for more details.
+errorLogsMessageGroupByInvalidFields=Spark program 'phase-1' failed with error: null.
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/resources/errorMessage.properties:32]: This error message signifies nothing and does not help in any kind of validation.

Comment (itsmekumari) [reply core-plugins/src/e2e-test/resources/errorMessage.properties:32]: There is no specific error shown in logs when providing invalid field/aggregates value set as macro. 
Screenshot 2024-12-10 at 11 30 17 AM
Screenshot 2024-12-10 at 11 28 09 AM

Comment (itsankit-google) [reply core-plugins/src/e2e-test/resources/errorMessage.properties:32]: Created a JIRA: https://cdap.atlassian.net/browse/PLUGIN-1828 to provide proper error message.

**Review by itsankit-google — APPROVED** [review]

File: core-plugins/src/e2e-test/resources/errorMessage.properties (hunk 2 above)
Comment (itsmekumari) [reply core-plugins/src/e2e-test/resources/errorMessage.properties:32]: Okay Thanks!

**Pull request merged**